dotnet/src/Connectors/Connectors.MistralAI.UnitTests/Client/MistralClientTests.cs

```diff
@@ -0,0 +1,542 @@
+
+        // Act
+        var executionSettings = new MistralAIPromptExecutionSettings { ToolCallBehavior = MistralAIToolCallBehavior.AutoInvokeKernelFunctions };
+        var chatHistory = new ChatHistory
+        {
+            new ChatMessageContent(AuthorRole.User, "What is the weather like in Paris?")
+        };
+        var response = await client.GetChatMessageContentsAsync(chatHistory, default, executionSettings, kernel);
+
+        // Assert
+        Assert.NotNull(response);
+        Assert.Single(response);
+        Assert.Equal("12°C\nWind: 11 KMPH\nHumidity: 48%\nMostly cloudy", response[0].Content);
+        Assert.Null(response[0].ModelId);
+        Assert.Equal(1, this.DelegatingHandler.SendAsyncCallCount);
+        Assert.Equal(3, chatHistory.Count);
+        Assert.Contains("GetWeather", invokedFunctions);
+    }
+
+    [Theory]
+    [InlineData("system", "System Content")]
+    [InlineData("user", "User Content")]
+    [InlineData("assistant", "Assistant Content")]
+    public void ValidateToMistralChatMessages(string roleLabel, string content)
+    {
+        // Arrange
+        using var httpClient = new HttpClient();
+        var client = new MistralClient("mistral-large-latest", httpClient, "key");
+        var chatMessage = new ChatMessageContent()
+        {
+            Role = new AuthorRole(roleLabel),
+            Content = content,
+        };
+
+        // Act
+        var messages = client.ToMistralChatMessages(chatMessage, default);
+
+        // Assert
+        Assert.NotNull(messages);
+        Assert.Single(messages);
+    }
+
+    [Fact]
+    public void ValidateToMistralChatMessagesWithFunctionCallContent()
+    {
+        // Arrange
+        using var httpClient = new HttpClient();
+        var client = new MistralClient("mistral-large-latest", httpClient, "key");
+        var content = new ChatMessageContent()
+        {
+            Role = AuthorRole.Assistant,
+            Items = [new FunctionCallContent("GetWeather"), new FunctionCallContent("GetCurrentTime")],
+        };
+
+        // Act
+        var messages = client.ToMistralChatMessages(content, default);
+
+        // Assert
+        Assert.NotNull(messages);
+        Assert.Single(messages);
+    }
+
+    [Fact]
+    public void ValidateToMistralChatMessagesWithFunctionResultContent()
+    {
+        // Arrange
+        using var httpClient = new HttpClient();
+        var client = new MistralClient("mistral-large-latest", httpClient, "key");
+        var content = new ChatMessageContent()
+        {
+            Role = AuthorRole.Tool,
+            Items = [new FunctionResultContent("12°C\nWind: 11 KMPH\nHumidity: 48%\nMostly cloudy"), new FunctionResultContent("15:20:44")],
+        };
+
+        // Act
+        var messages = client.ToMistralChatMessages(content, default);
+
+        // Assert
+        Assert.NotNull(messages);
+        Assert.Equal(2, messages.Count);
+    }
+
+    public sealed class WeatherPlugin
```

Comment:
Just wondering, do we already support description on the plugin level?
```suggestion
    [Description]
    public sealed class WeatherPlugin
```

Reply:
No not yet